File: ui/src/app/items/services/items/incident-stub.ts
export class ItemStub {
  public type: string;
  public id: string;
  public incident_number: string;
  public creation_date: string;
  public lastupdated_date: string;
  public institution: string;
  public consultant: string;
  public access_rights: string[];
  public totalItemsCount?: number;
  public responsible_institution?: string;
  public responsible_institution_others?: string;

  constructor(values: any) {
    this.type = values.type;
    this.id = values.id;
    this.incident_number = values.incident_number;
    this.creation_date = values.creation_date;
    this.lastupdated_date = values.lastupdated_date ? values.lastupdated_date : this.creation_date;
    this.institution = values.institution;
    this.consultant = values.consultant;
    this.access_rights = values.access_rights;
    this.totalItemsCount = values.totalItemsCount;
    this.responsible_institution = values.responsible_institution;
    this.responsible_institution_others = values.responsible_institution_others;
  }
}

File: app/modelHelper/IncidentStub.java
package modelHelper;

import models.UserContext;
import org.joda.time.Instant;

public class IncidentStub {
    public long id;
    public String type;
    public IncidentMetadata metadata = new IncidentMetadata();

    public static IncidentStub createInitialIncident(String type, long incidentNumber, Instant created, UserContext userContext) {
        IncidentStub incidentStub = new IncidentStub();
        incidentStub.id = incidentNumber;
        incidentStub.type = type.toUpperCase();
        incidentStub.metadata.incident_number = String.valueOf(incidentNumber);
        incidentStub.metadata.creation_date = created.toString();
        incidentStub.metadata.lastupdated_date = created.toString();
        incidentStub.metadata.institution = userContext.getInstitution();
        incidentStub.metadata.responsible_institution = userContext.getInstitutionKey();
        incidentStub.metadata.external_incident = false;
        incidentStub.metadata.consultant = userContext.getName();
        return incidentStub;
    }
}
